amf-shapes/shared/src/main/scala/amf/shapes/client/platform/model/domain/jsonldinstance/JsonLDScalar.scala

```diff
@@ -0,0 +1,18 @@
+package amf.shapes.client.platform.model.domain.jsonldinstance
+
+import amf.core.client.scala.model.domain.AmfScalar
+import amf.shapes.client.platform.model.domain.jsonldinstance.{JsonLDScalar => InternalJsonLDScalar}
+
+import scala.scalajs.js.annotation.{JSExportAll, JSExportTopLevel}
+
+@JSExportAll
+class JsonLDScalar(private [amf] val _internal: InternalJsonLDScalar, val value: Any, val dataType: String ) extends AmfScalar(value) whit JsonLDElement{
+
+
+  @JSExportTopLevel("JsonLDScalar")
+  def this() = this(InternalJsonLDScalar())
```

Comment:
There is not empty constructor for the internal JsonLDScalar, it always requires model and datatype parameters (could check the use of internal). The platform constructors must follow the internal ones.